### DISCUSSION TITLE
Add `repository.url` to the `package.json` so alpinejs is deployable as webjar

### DISCUSSION THREAD

**Discussion by wimdeblauwe** [alpinejs/alpine#3147] [3. Feature ideas]
Webjars is a way to use NPM packages in Java based projects. I use this in my Java/Spring Boot/Thymeleaf projects. It used to be possible to wrap alpinejs in a webjar up to version 2.8.2 of alpinejs, but now it is no longer possible.

The webjar automatic wrapper creation process requires a `repository.url` in the `package.json` and that seems to be no longer there. Would it be possible to add this back in so we can create webjars of alpinejs easily to use alpinejs in our Java projects?

See https://github.com/webjars/webjars/issues/1960 for some more info.

**Comment by ekwoka** [comment]
Not sure what a awebjar is, but more compatibility is always good.

One thing that's unclear in that discussion is whether a folder path for a repo url is valid.

This is a monorepo, so each package.json would need to point to a folder in the tree, not the root repo.

I have an open PR making adjustments to all the package.json that I can try adding all this stuff to anyway and see if that works.

I have my own projects in a monorepo so I might be able to test if this plan would enable webjar deployment.

**Reply by wimdeblauwe** [reply]
Do you mean having `/something` as repo url or having `http://github.com/alpinejs/something` as repo url?

**Reply by ekwoka** [reply]
the second.

It would need to be: `https://github.com/alpinejs/alpine/tree/main/packages/alpinejs`

Looks liek the correct way to do it following NPMs spec is like

```json
{
  "repository": {
    "type": "git",
    "url": "https://github.com/alpinejs/alpine.git",
    "directory": "packages/alpinejs"
  }
}
```

I can add that to the PR, and just see if that works.

**Reply by wimdeblauwe** [reply]
ok, it is worth trying I think. Not sure what exactly webjars does with the url. There would need to be a new version with this published on npm to be able to try it.

**Reply by ekwoka** [reply]
Added to https://github.com/alpinejs/alpine/pull/3112

**Reply by wimdeblauwe** [reply]
got reply from the person that wrote webjars and it should be fine like this.
